akka-stream/src/main/scala/akka/stream/IOResult.scala

```diff
@@ -12,7 +16,11 @@ import scala.util.{ Failure, Success, Try }
  * @param count Numeric value depending on context, for example IO operations performed or bytes processed.
  * @param status Status of the result. Can be either [[akka.Done]] or an exception.
  */
-final case class IOResult private[stream] (count: Long, status: Try[Done]) {
+final class IOResult private (val count: Long, val status: Try[Done])
```

Comment:
Making the constructor public doesn't require removal of case class, or do I miss something? 

Why pay the price of complexity before we have to? Best would of course have been if it wasn't a case class to begin with, but now it is.